Module Class Refactor

stanbar opened the pull request:
replace `check()` with `validateInput()` which must throw `RequiredInputNotFound` if the input is not present or incorrect.
add public wrapper methods `validateAndEncrypt()` which will force framework to `validateInputs()` before en/decryption
modules override protected methods decrypt/encrypt
implemented changes in `PinModule`, `ButtonModule` and `ExampleModule`

@PatrykMilewski please approve this concept and implement those changes in `AuthorizationServerModule`

Closes #87 

server port as config property
default port 80
default secure-port: 433

Took 1 hour 59 minutes

stanbar committed:
replace check() with validateInput() which must throw if the input is not present or incorrect.
add public wrapper methods validateAndEncrypt() which will force framework to validateInputs before en/decryption
modules override protected methods decrypt/encrypt
implemented changes in Pin, Button and Example Modules

Took 53 minutes

stanbar committed:
input keys as constants

Took 7 minutes

PatrykMilewski commented:
I've been looking into code and check was called in manager before inputs were provided. By doing it like this, we have to make changes in Manager class (in code: for each module call check)

PatrykMilewski commented:
Actually I think that changing check to validateInput is correct and logically correct, but still there is need of Manger refactor because of this

stanbar committed:
fixed typo in config.yaml "accept-unstrusted-certs" => "accept-untrusted-certs"
fixed tests

Took 5 minutes

stanbar commented:
> I've been looking into code and check was called in manager before inputs were provided. By doing it like this, we have to make changes in Manager class (in code: for each module call check)

Where ? I removed `check()` from `Module` and everything works. I even started the program without any problems.

PatrykMilewski commented:
Please fix auth server tests

PatrykMilewski commented:
They were using check method

stanbar commented:
Now you just need to move the 
```
    public boolean check() {
        if (hasInput(PASSWORD)) {
            try {
                String password = getInput(PASSWORD);
                serverCredentials = new Credentials(walletUUID.toString(),
                        Base64.getUrlEncoder().encodeToString(password.getBytes()));
                initialize();
            } catch (RequestException e) {
                return false;
            }
            return true;
        } else return serverCredentials != null;
    }
```
method body because I'm not sure where it should go

PatrykMilewski commented:
> Now you just need to move the
> 
> ```
>     public boolean check() {
>         if (hasInput(PASSWORD)) {
>             try {
>                 String password = getInput(PASSWORD);
>                 serverCredentials = new Credentials(walletUUID.toString(),
>                         Base64.getUrlEncoder().encodeToString(password.getBytes()));
>                 initialize();
>             } catch (RequestException e) {
>                 return false;
>             }
>             return true;
>         } else return serverCredentials != null;
>     }
> ```
> method body because I'm not sure where it should go

Okay, uunderstood

stanbar committed:
server port as config property
default port 80
default secure-port: 433

Took 22 minutes

PatrykMilewski committed:
Continuation of Module class refactor

Changed methods validateAndEncrypt/Decrypt to just encrypt / decrypt

Removed register method

Added a new exception ModuleInitializationException

Made AuthServer constructors try to initialize logic

Moved more reposibility to encrypt / decrypt methods

Took 53 minutes

PatrykMilewski committed:
Merge remote-tracking branch 'origin/module-class-refactor' into module-class-refactor

stanbar committed:
enforce validateInputs by public api de/encryptKeyPart
passing tests

Took 44 minutes

stanbar committed:
fix tests

Took 15 minutes
